spec/unit/testing/database_url_spec.rb

```diff
@@ -3,24 +3,77 @@
 RSpec.describe Hanami::DB::Testing do
   subject { described_class.method(:database_url) }
 
-  it "transforms _dev to _test" do
-    expect(subject.call("/bookshelf_dev")).to eq "/bookshelf_test"
+  shared_examples "URL Transforms" do |scheme|
+    define_method(:url) { |path| URI.join("#{scheme}://localhost", path).to_s }
+
+    it "transforms _dev to _test" do
+      expect(subject.call(url("/bookshelf_dev"))).to eq url("/bookshelf_test")
+    end
+
+    it "transforms _development to _test" do
+      expect(subject.call(url("/bookshelf_development"))).to eq url("/bookshelf_test")
+    end
+
+    it "does not transform _test" do
+      expect(subject.call(url("/bookshelf_test"))).to eq url("/bookshelf_test")
+    end
+
+    it "appends to non-conforming paths" do
+      expect(subject.call(url("/bookshelf_database"))).to eq url("/bookshelf_database_test")
+    end
+
+    it "accepts any #to_s object such as URI" do
+      url = URI("#{scheme}://localhost:5432/bookshelf_development")
+      expect(subject.call(url)).to eq "#{scheme}://localhost:5432/bookshelf_test"
+    end
   end
 
-  it "transforms _development to _test" do
-    expect(subject.call("/bookshelf_development")).to eq "/bookshelf_test"
+  context "postgres scheme" do
+    include_examples "URL Transforms", :postgres
+
+    it "preserves query params" do
+      url = "postgres://user:pass@/bookshelf_development?host=/var/run/postgresql/.s.PGSQL.5432"
+      expect(subject.call(url)).to eq "postgres://user:pass@/bookshelf_test?host=/var/run/postgresql/.s.PGSQL.5432"
```

Comment:
This is a fun edge case. Running migrations in Google Cloud Build generally requires using [exec-wrapper](https://github.com/GoogleCloudPlatform/ruby-docker/tree/master/app-engine-exec-wrapper) because you can't run multiple docker containers simultaneously.

For boring reasons not worth getting into, you can't access the db port from within the internal image, so you have to resort to the unix socket instead. But you can't use a file path in the URI path, so it's important that these query values are persisted.

Reply:
Wow, I'm glad you've _Seen Things_ enough to account for this!